android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/shopping/GeckoShoppingProductAnalysis.kt

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.engine.gecko.shopping
+
+import mozilla.components.concept.engine.shopping.ShoppingProductAnalysis
+
+/**
+ * Holds the result of the analysis of a shopping product.
+ *
+ * @property productId Product identifier (ASIN/SKU)
+ * @property analysisURL Analysis URL
+ * @property grade Reliability grade for the product's reviews
+ * @property adjustedRating Product rating adjusted to exclude untrusted reviews
+ * @property needsAnalysis Boolean indicating if the analysis is stale
+ * @property lastAnalysisTime Time since the last analysis was performed
+ * @property deletedProductReported Boolean indicating if reported that this product has been deleted
+ * @property deletedProduct Boolean indicating if this product is now deleted
+ * @property highlights Object containing highlights for product
+ */
+data class GeckoShoppingProductAnalysis(
+    override val productId: String,
+    val analysisURL: String?,
+    val grade: String?,
+    val adjustedRating: Double,
```

Comment:
Can this be null as well?

Same question for `lastAnalysisTime` and both `deletedX` properties

Reply:
In Java `Double` is a primitive value and is not nullable. GV is written in Java, so you will not get a `null` from GV.

Reply:
Sorry, I was wrong above - `Double` in Java is an object (not the primitive `double`), but in GV when we get it from the Gecko's response - we check for null, and substitute `0.0` instead of `null`. So the prop is annotated `@NonNull` in GV code. You cannot get `null` here

Reply:
(but the `lastAnalysisTime` and the `deleted` are indeed primitive types, so not nullable)

Reply:
I think this would lead to double checking across the layers, GV is checking if `null` and then setting `0.0` and the AC is just passing it through and then Fenix UI layer does the same check but for `0.0`, lots of translations and it can be simplified with having it nullable. I think we should revisit that and have a discussion about it. 

By just keeping it null, it simplifies and we don't get into a state where we check for `0.0` or empty string ever.